src/overlays/actors/ovl_En_Bigslime/z_en_bigslime.c

```diff
@@ -54,289 +243,2911 @@ const ActorInit En_Bigslime_InitVars = {
+    Vec3f dustPos;
+    Vec3f hahenVel;
+    s16 yaw = func_800DFCDC(GET_ACTIVE_CAM(globalCtx));
+    s16 yawReverse = yaw + 0x8000;
+    s32 i;
+
+    this->gekkoCollider.base.ocFlags1 &= ~OC1_ON;
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_MINIFROG, this->actor.world.pos.x, this->actor.world.pos.y,
+                this->actor.world.pos.z, 0, yawReverse, 0, this->actor.params);
+
+    dustPos.x = (Math_SinS(yawReverse) * 20.0f) + this->actor.world.pos.x;
+    dustPos.y = this->actor.world.pos.y + 20.0f;
+    worldPos = &this->actor.world.pos;
+    dustPos.z = (Math_CosS(yawReverse) * 20.0f) + this->actor.world.pos.z;
+
+    Audio_PlaySoundAtPosition(globalCtx, worldPos, 40, NA_SE_EN_NPC_APPEAR);
+
+    // dust cloud where the red frog appears
+    func_800B0DE0(globalCtx, &dustPos, &gZeroVec3f, &gZeroVec3f, &sDustPrimColor, &sDustEnvColor, 500, 50);
+
+    for (i = 0; i < 25; i++) {
+        hahenVel.x = randPlusMinusPoint5Scaled(5.0f);
+        hahenVel.y = Rand_ZeroFloat(3.0f) + 4.0f;
+        hahenVel.z = randPlusMinusPoint5Scaled(5.0f);
+        EffectSsHahen_Spawn(globalCtx, worldPos, &hahenVel, &sHahenAccel, 0, Rand_S16Offset(12, 3),
+                            HAHEN_OBJECT_DEFAULT, 10, 0);
+    }
+
+    this->spawnFrogTimer = 40;
+    Math_Vec3f_Diff(&subCam->eye, &subCam->at, &this->subCamDistToFrog);
+    this->actionFunc = EnBigslime_FrogSpawn;
+}
+
+void EnBigslime_FrogSpawn(EnBigslime* this, GlobalContext* globalCtx) {
+    Camera* subCam = Play_GetCamera(globalCtx, this->subCamId);
+    Vec3f subCamEye;
+    f32 subCamZoom;
+
+    this->spawnFrogTimer--;
+
+    // Zoom the camera in and out at the newly spawned red frog
+    subCamZoom = sin_rad(this->spawnFrogTimer * (M_PI / 5)) * ((0.04f * (this->spawnFrogTimer * 0.1f)) + 0.02f) + 1.0f;
+    subCamEye.x = subCam->at.x + (this->subCamDistToFrog.x * subCamZoom);
+    subCamEye.z = subCam->at.z + (this->subCamDistToFrog.z * subCamZoom);
+    subCamEye.y = subCam->at.y + (this->subCamDistToFrog.y * subCamZoom);
+    Play_CameraSetAtEye(globalCtx, this->subCamId, &subCam->at, &subCamEye);
+
+    if (this->spawnFrogTimer == 0) {
+        EnBigslime_EndCutscene(this, globalCtx);
+        EnBigslime_SetupDespawn(this);
+    }
+}
+
+void EnBigslime_SetupDespawn(EnBigslime* this) {
+    s32 i;
+
+    for (i = 0; i < MINISLIME_NUM_SPAWN; i++) {
+        this->minislime[i]->actor.params = MINISLIME_DEFEAT_MELT;
+    }
+
+    this->isDespawned = 0;
+    this->gekkoCollider.base.ocFlags1 &= ~OC1_ON;
+    this->actionFunc = EnBigslime_Despawn;
+}
+
+void EnBigslime_Despawn(EnBigslime* this, GlobalContext* globalCtx) {
+    s32 i;
+    s32 counter = 0;
+
+    for (i = 0; i < MINISLIME_NUM_SPAWN; i++) {
+        if (this->minislime[i]->actor.shape.shadowAlpha == 0) {
+            counter++;
+        }
+    }
+
+    if (this->isDespawned == 0) {
+        Actor_SetRoomClearedTemp(globalCtx, globalCtx->roomCtx.currRoom.num);
+        this->isDespawned = 1;
+    }
```

Comment:
This variable looks like a boolean
```suggestion
    if (!this->isDespawned) {
        Actor_SetRoomClearedTemp(globalCtx, globalCtx->roomCtx.currRoom.num);
        this->isDespawned = true;
    }
```